### DISCUSSION TITLE
how to push large projects

### DISCUSSION THREAD

**Discussion by Daniah02** [community/community#142166] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hi, I'm Dania and am having trouble pushing my large project to my repository from VS code, each time I try to push around 5K changes, an error occurs saying that the project is too large. I’m wondering if the solution could be by just subscribing for a plan that accommodate larger repository or is the pushing process have some kind of limits or is there any other solution that I can use to push 3 GB project to my repository.

**Comment by KamarulAdha** [comment]
Hola!

Without reviewing your commits and files, the issue is likely not due to the number of changes but rather the size of the files you're trying to commit.

If you're pushing code, GitHub should handle it fine, regardless of the number of changes. However, if your push includes large files like node_modules, static assets, datasets, zip files, or other large binaries, this could trigger an error, as GitHub imposes limits on repository size and individual file sizes.

I recommend setting up a .gitignore file to exclude large or unnecessary files (like graphics, binaries, or build artifacts) from being pushed to your repository. You should only push the code and essential project files. For large files, you could consider storing them in external storage services like an Amazon S3 bucket or Hugging Face datasets.

Let me know if this was helpful!
